src/ui-kit/experimental/hierarchical/hierarchical-tree/hierarchical-tree.component.spec.ts

```diff
@@ -134,17 +135,90 @@ export class HierarchicalDataService implements SamHiercarchicalServiceInterface
     this.loadedData = data;
   }
 
-  getDataByText(currentItems: number, searchValue?: string): Observable<SearchByTextResult> {
-    return null;
+  getDataByText(currentItems: number, searchValue?: string): Observable<SamHiercarchicalServiceResult> {
+    let itemIncrease = 25;
```

Comment:
@thomastighe I don't think this is really an issue, but this is a lot of code (in one function) to setup for a test.

https://dzone.com/articles/unit-testing-smells-what-are-your-tests-telling-yo

I don't think there's anything funky going on here, its just one of those things that jumps out to me that something could be weird in the design if the tests are this big. I think it is actually fine, and this is just a case of really needing a mock to run the test.

I think as long as you make sure you're sending enough test cases to cover enough variation in the method params, everything is good.

Reply:
I meant for this comment to apply to the couple of other mocked methods as well. The one I would really try to cover with enough test cases is the one with 4 method params. This one is pretty simple. 

And I only mean to cover it with test cases of realistic values that might be sent to this from our components that call this method. The output of these methods should depend on the implementation, so its not in our area of concern if the output of this is ultimately incorrect.

Reply:
Ok?